### PR TITLE
[Telegram]: added telegram bot for sending notifications

### PR DESCRIPTION
**Pull request by TarasYurkovskyi**
### Added:

- TelegramBot class
- BotInitializer class
- New dependencies for telegram
- Booking repository
- Notification service interface
- TelegramNotificationService

### Changed:

- Add new column for user model
- Change script for creating user table
- Add new methods to user repository

### CODE REVIEW

File: src/main/resources/db/changelog/changes/01-create-users-table.yaml [hunk 1]
```diff
@@ -12,6 +12,11 @@ databaseChangeLog:
                   autoIncrement: true
                   constraints:
                     primaryKey: true
+              - column:
+                  name: chat_Id
```
Comment (vadymhrnk) [inline src/main/resources/db/changelog/changes/01-create-users-table.yaml:16]: I think it's better to have consistent naming and the version in model looks better to me
```
@Column(name = "chat_id", unique = true)
private Long chatId;
```

File: src/main/java/com/example/accommodationbookingapp/telegram/BotInitializer.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    private final TelegramBot bot;
+
+    @EventListener({ContextRefreshedEvent.class})
+    public void init() throws TelegramApiException {
+        TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
+        try {
+            telegramBotsApi.registerBot(bot);
+        } catch (TelegramApiException e) {
+            throw new RuntimeException("Cannot initialize the Bot" + e.getMessage());
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/telegram/BotInitializer.java:23]: Better to put this text in constant field

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,7 +1,7 @@
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.datasource.url=jdbc:mysql://localhost:3306/${DATABASE}
-spring.datasource.username=${USERNAME}
-spring.datasource.password=${PASSWORD}
+spring.datasource.url=jdbc:mysql://localhost:3306/hibernate
+spring.datasource.username=root
+spring.datasource.password=DoinTheMost2005!
```
Comment (vadymhrnk) [inline src/main/resources/application.properties:4]: Better to remove this and use environment variable in IDE

**Review by vadymhrnk — CHANGES_REQUESTED** [review]
Well done! Small points for you to consider

**Review by vadymhrnk — APPROVED** [review]
LGTM

**Review by annton-shakur — APPROVED** [review]
Well done!

File: src/main/java/com/example/accommodationbookingapp/telegram/BotInitializer.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    private static final String INIT_ERR = "Cannot initialize the Bot";
+    private final TelegramBot bot;
+
+    @EventListener({ContextRefreshedEvent.class})
+    public void init() throws TelegramApiException {
+        TelegramBotsApi telegramBotsApi = new TelegramBotsApi(DefaultBotSession.class);
+        try {
+            telegramBotsApi.registerBot(bot);
+        } catch (TelegramApiException e) {
+            throw new RuntimeException(INIT_ERR + e.getMessage());
```
Comment (Sofasmile) [inline src/main/java/com/example/accommodationbookingapp/telegram/BotInitializer.java:23]:
```suggestion
            throw new RuntimeException(INIT_ERR + e.getMessage(), e);
```

File: src/main/java/com/example/accommodationbookingapp/telegram/TelegramBot.java [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+        this.bookingRepository = bookingRepository;
+    }
+
+    @Override
+    public String getBotUsername() {
+        return "YourBotUsername";
+    }
+
+    @Override
+    public void onUpdateReceived(Update update) {
```
Comment (Sofasmile) [inline src/main/java/com/example/accommodationbookingapp/telegram/TelegramBot.java:92]: Recommendation: try to split this method and use strategy

**Review by MinoDentori — APPROVED** [review]
Looks good for me

**Pull request merged**